manifests/monitored_nfs_mount.pp

```diff
@@ -0,0 +1,39 @@
+# Copyright (c) 2018 The Regents of the University of Michigan.
+# All Rights Reserved. Licensed according to the terms of the Revised
+# BSD License. See LICENSE.txt for details.
+
+# nebula::monitored_nfs_mount
+#
+# Configure an NFS mount and monitor it via monitor.pl
+#
+# @example
+#   nebula::monitored_nfs_mount { 'Test::More': }
+define nebula::monitored_nfs_mount(
+  String $device,
+  Optional[String] $options = 'auto,hard',
+  Boolean $smartconnect = false,
+) {
+  ensure_packages(['nfs-common'], {'ensure' => 'present'})
+
+  file { $title:
+    ensure => 'directory',
+  }
+
+  mount { $title:
+    ensure  => 'mounted',
+    options => $options,
+    fstype  => 'nfs',
+  }
+
+  if($smartconnect) {
+    Mount[$title] {
+      require => ['Package[nfs-common]','File[/etc/resolv.conf]'],
+      tag     => 'private_network'
+    }
+  }
```

Comment:
otherwise should still require `Package[nfs-common]`

Reply:
should consider specifying these requirements over in the smartconnect profile and just tagging the monitored nfs mounts